pavelib/servers.py

```diff
@@ -2,24 +2,28 @@
 Run and manage servers for local development.
 """
 from __future__ import print_function
-import sys
 import argparse
 from paver.easy import *
+
+from .assets import collect_assets
 from .utils.cmd import django_cmd
 from .utils.process import run_process, run_multi_processes
 
 
 DEFAULT_PORT = {"lms": 8000, "studio": 8001}
 DEFAULT_SETTINGS = 'devstack'
+OPTIMIZED_SETTINGS = "devstack_optimized"
+OPTIMIZED_ASSETS_SETTINGS = "test_static_optimized"
 
 
-def run_server(system, settings=None, port=None, skip_assets=False, contracts=False):
+def run_server(system, settings=None, asset_settings=None, collect_static=False, port=None, no_contracts=False):
     """
     Start the server for the specified `system` (lms or studio).
     `settings` is the Django settings module to use; if not provided, use the default.
+    `collect_static` defaults to False, but if True then static files will be collected.
+    `asset_settings` is the settings to use when generating assets; if not provided, assets are not generated.
```

Comment:
The asset_settings are used for more than just collectstatic. They are used for generating SASS and coffee, amongst other asset handling. Collecting the static files for production is only one part of what update_assets does.

The purpose of updating assets without collecting static is when you know that the server you are starting isn't going to use the collected static assets. In general, if you are using devstack then there's no need to collect static, but if you use a different settings file then you might want to. Currently the only settings file that needs the collected assets is my new devstack_optimized, but I didn't want to assume that that was the case.

I should update the documentation to make this all clearer.